### DISCUSSION TITLE
`$WEZTERM_PANE` when connected over SSH?

### DISCUSSION THREAD

**Discussion by mrjones2014** [wezterm/wezterm#5378] [Q&A] [answered]
### What Operating System(s) are you running on?

Linux X11

### Which Wayland compositor or X11 Window manager(s) are you using?

GNOME's default, which I think is Mutter?

### WezTerm version

wezterm 20240203-110809-5046fc22

### Ask your question!

I've been managing my home server via SSH and have been running into this issue with my smart-splits.nvim Neovim plugin: https://github.com/mrjones2014/smart-splits.nvim/issues/163

Basically, the plugin sets a Wezterm user var `IS_NVIM` on load and unsets it on nvim exit, however, when connected over SSH, the `$WEZTERM_PANE` variable isn't set on the remote shell, so the plugin doesn't work because it tries to run stuff like `wezterm activate-pane-direction Up`, but this fails if `$WEZTERM_PANE` isn't set and you don't pass `--pane-id`.

Is there a good way to work around this issue?

**Comment by wez** [comment, marked as answer]
WEZTERM_PANE is really intended for consumption by a local multiplexer and tools.  If it was set on a remote host, wezterm expects that it is valid for a multiplexer running on that same host, not the client host.

In wezterm's ssh sessions we try to set WEZTERM_REMOTE_PANE but most sshd configurations prohibit setting arbitrary environment variables by default, so it is not usually set.

Outside of using wezterm's multiplexer to get to the remote host, I don't think there is a good way to communicate this sort of stuff back from a remote host.

**Reply by mrjones2014** [reply]
Thanks for the explanation. It sounds like the only way to make this work is to set `AcceptEnv WEZTERM_REMOTE_PANE` in the servers SSH config.
